adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala

```diff
@@ -380,53 +387,210 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord])
+            // move the head file back - essentially, unroll the prep for concat
+            fs.rename(newHeadPath, headPath)
+
+            // the only reason we have this try/catch is to unwrap the wrapped
+            // exception and rethrow. this gives clearer logging messages
+            throw ite.getTargetException
+          }
+        }
 
-      FileUtil.copyMerge(fs, tailPath, fs, filePath, true, conf, null)
-      fs.delete(tailPath, true)
+        // move concated file
+        fs.rename(newHeadPath, outputPath)
 
-    }
+        // delete tail files
+        fs.delete(tailPath, true)
+      } catch {
+        case e: Throwable => {
 
-  }
+          log.warn("Caught exception when merging via Hadoop FileSystem API:\n%s".format(e))
+          log.warn("Retrying as manual copy from the driver which will degrade performance.")
+
+          // doing this correctly is surprisingly hard
+          // specifically, copy merge does not care about ordering, which is
+          // fine if your files are unordered, but if the blocks in the file
+          // _are_ ordered, then hahahahahahahahahaha. GOOD. TIMES.
+          //
+          // fortunately, the blocks in our file are ordered
+          // the performance of this section is hilarious
+          // 
+          // specifically, the performance is hilariously bad
+          //
+          // but! it is correct.
+
+          // open our output file
+          val os = fs.create(outputPath)
+
+          // prepend our header to the list of files to copy
+          val filesToCopy = Seq(headPath) ++ tailFiles.toSeq
+
+          // here is a byte array for copying
+          val ba = new Array[Byte](1024)
 
-  implicit def str2Path(str: String): Path = new Path(str)
+          @tailrec def copy(is: InputStream,
+                            os: OutputStream) {
+
+            // make a read
+            val bytesRead = is.read(ba)
+
+            // did our read succeed? if so, write to output stream
+            // and continue
+            if (bytesRead >= 0) {
+              os.write(ba, 0, bytesRead)
+
+              copy(is, os)
+            }
+          }
+
+          // loop over allllll the files and copy them
+          val numFiles = filesToCopy.length
+          var filesCopied = 1
+          filesToCopy.foreach(p => {
+
+            // print a bit of progress logging
+            log.info("Copying file %s, file %d of %d.".format(
+              p.toString,
+              filesCopied,
+              numFiles))
+
+            // open our input file
+            val is = fs.open(p)
+
+            // until we are out of bytes, copy
+            copy(is, os)
+
+            // close our input stream
+            is.close()
+
+            // increment file copy count
+            filesCopied += 1
+          })
+
+          // flush and close the output stream
+          os.flush()
+          os.close()
+
+          // delete temp files
+          //fs.delete(headPath, true)
```

Comment:
Both are removed here... Although, I commented those lines out when debugging something; let me fix that real quick.

Are you referring to the code several lines above at https://github.com/bigdatagenomics/adam/pull/917/files#diff-a0df6432a6d340425e3510af17cb8bd2R516 ? The headPath file there doesn't need to be removed because the `concat` merges all of the tail files _into_ the head file. We then move the head file to the desired output path.